src/adera/SysExhaustPlume.cpp

```diff
@@ -39,52 +41,135 @@
 
 using namespace adera::active;
 
-using osp::AssetImporter;
-
-using osp::DependRes;
-using osp::Package;
-
 using osp::active::ActiveScene;
 using osp::active::ActiveEnt;
 
+/**
+ * Attach a visual exhaust plume effect to MachineRocket
+ *
+ * Searches the hierarchy under the specified MachineRocket entity and
+ * attaches an ACompExhaustPlume to the rocket's plume node. A graphical
+ * exhaust plume effect will be attached to the node by SysExhaustPlume
+ * when it processes the component.
+ *
+ * @param rScene [ref] Scene containing the following entities
+ * @param part [in] Entity containing a plume in its descendents
+ * @param mach [in] Entity containing MachineRocket
+ */
+void attach_plume_effect(ActiveScene &rScene, ActiveEnt part, ActiveEnt mach)
+{
+    using osp::active::ACompName;
+    using osp::active::EHierarchyTraverseStatus;
+    using osp::active::SysHierarchy;
 
-using osp::active::ACompMesh;
-using osp::active::ACompShader;
-using osp::active::ACompTransparent;
-using osp::active::ACompVisible;
+    ActiveEnt plumeNode = entt::null;
 
-void SysExhaustPlume::initialize_plume(ActiveScene& rScene, ActiveEnt node)
-{
-    using Magnum::GL::Mesh;
-    using Magnum::GL::Texture2D;
-    using Magnum::Trade::ImageData2D;
-    using namespace Magnum::Math::Literals;
-    using adera::shader::PlumeShader;
+    auto findPlumeHandle = [&rScene, &plumeNode](ActiveEnt ent)
+    {
+        static constexpr std::string_view namePrefix = "fx_plume_";
+        auto const* name = rScene.reg_try_get<ACompName>(ent);
 
-    auto const& plume = rScene.reg_get<ACompExhaustPlume>(node);
+        if (nullptr == name)
+        {
+            return EHierarchyTraverseStatus::Continue;
+        }
+
+        if (0 == name->m_name.compare(0, namePrefix.size(), namePrefix))
```

Comment:
https://en.cppreference.com/w/cpp/string/basic_string/starts_with
https://en.cppreference.com/w/cpp/string/basic_string_view/starts_with ?

Reply:
These are new in C++20 only

Reply:
really? that's obnoxious. My bad.